docs/proposals/06-immutable-etcd-backups.md

```diff
@@ -0,0 +1,292 @@
+
+## Risks and Mitigations
+
+- **Increased Storage Costs:**
+
+  - **Risk:** Introducing immutability could lead to increased storage costs due to the inability to delete backups before the immutability period ends.
+  - **Mitigation:** Operators should carefully configure immutability periods and monitor storage utilization. Garbage collection will help mitigate long-term storage growth.
+
+- **Backup Gaps During Hibernation:**
+
+  - **Risk:** Hibernated clusters might not receive necessary backups, potentially leading to compliance issues.
+  - **Mitigation:** The compaction job's enhancement ensures that backups are taken during hibernation at configured intervals.
+
+- **Excluding Critical Snapshots:**
+
+  - **Risk:** Excluding snapshots during restoration might be misused or lead to incomplete data restoration.
+  - **Mitigation:** Restrict the ability to tag snapshots for exclusion to authorized personnel. Implement audit logging for actions that tag snapshots.
+
+## Operational Considerations
+
+Operators need to:
+
+- **Bucket Configuration:**
+
+  - Configure buckets with appropriate immutability settings before deploying ETCD clusters.
+  - Ensure that the immutability periods align with organizational policies.
+
+- **Compaction Job Configuration:**
+
+  - Set the `--hibernation-snapshot-interval` flag according to the desired snapshot frequency during hibernation.
+  - Monitor compaction jobs and logs for any issues.
+
+## Alternatives
+
+### Object-Level Immutability Policies vs. Bucket-Level Immutability Policies
+
+An alternative to implementing immutability via bucket-level immutability policies is to use object-level immutability policies. Object-level immutability allows for more granular control over the immutability periods of individual objects within a bucket, whereas bucket-level immutability applies a uniform immutability period to all objects in the bucket.
+
+#### Feasibility Study: Immutable Backups on Cloud Providers
+
+Major cloud storage providers such as Google Cloud Storage (GCS), Amazon S3, and Azure Blob Storage (ABS) support both bucket-level and object-level immutability mechanisms to enforce data immutability.
+
+1. **Bucket-Level Immutability Policies:**
+
+     - **Applies Uniformly:** Applies a uniform immutability period to all objects within a bucket.
+     - **Immutable Objects:** Once set, objects cannot be modified or deleted until the immutability period expires.
+     - **Simplified Management:** Simplifies management by applying the same policy to all objects.
+
+2. **Object-Level Immutability Policies:**
+
+     - **Granular Control:** Allows setting immutability periods on a per-object basis.
+     - **Flexible Immutability Durations:** Offers granular control, enabling different immutability durations for individual objects.
+     - **Varying Requirements:** Can accommodate varying immutability requirements for different types of backups.
+
+#### Considerations for Object-Level Immutability
```

Comment:
We should link this in the Overview as an alternate which was considered and can be read here, but there we only talk about our choosen approch.

Reply:
Outdated. This is already present.